Dockerfile

```diff
@@ -74,13 +74,15 @@ RUN set -ex; \
             python3-dev \
             apache-arrow-dev \
             py3-pip; \
+        # pgai requires pip 23.0.1 or greater due to the use of --break-system-packages flag
+        if [ "$(pip --version | awk '{print $2; exit}')" \< "23.0.1" ]; then \
+            python3 -m pip install --upgrade pip==23.0.1; \
+        fi; \
         git clone --branch ${PGAI_VERSION} https://github.com/timescale/pgai.git /build/pgai; \
         cd /build/pgai; \
-        # note: this is a hack. pyarrow will be built from source, so must be pinned to this arrow version \
-        echo pyarrow==$(pkg-config --modversion arrow) >> ./projects/extension/requirements.txt; \
         if [ "$TARGETARCH" == "386" ]; then \
             # note: pinned because pandas 2.2.0-2.2.3 on i386 is affected by https://github.com/pandas-dev/pandas/issues/59905 \
-            echo pandas==2.1.4 >> ./projects/extension/requirements.txt; \
+            sed -i '/dependencies= \[/a \    "pandas==2.1.4",' projects/extension/pyproject.toml; \
```

Comment:
Ended up adding the dependency to the `pyproject.toml` file as `pip install` is not considering `requirements-lock.txt`.

Reply:
An alternative option (which doesn't require hacking around in the pyproject.toml) is to create a [`contstraints.txt`](https://pip.pypa.io/en/latest/cli/pip_install/#cmdoption-c) file and get pip to use that (by setting the `PIP_CONSTRAINT` env var):
```
echo "pandas==2.1.4" > constraints.txt
export PIP_CONSTRAINT=$(pwd)/constraints.txt
```

Reply:
> An alternative option (which doesn't require hacking around in the pyproject.toml) is to create a [`contstraints.txt`](https://pip.pypa.io/en/latest/cli/pip_install/#cmdoption-c) file and get pip to use that (by setting the `PIP_CONSTRAINT` env var):
> 
> ```
> echo "pandas==2.1.4" > constraints.txt
> export PIP_CONSTRAINT=$(pwd)/constraints.txt
> ```

TIL! 🧠 

I actually like your solution with constraints more than the hack because it exactly does what we need. 
Changing it right now. Thanks for such a great suggestion @JamesGuthrie

Reply:
YIL (yesterday I learned!) I also like it more than what we did before.